frappe/integrations/doctype/nextcloud_settings/nextcloud_settings.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+from frappe import _
+from frappe.utils.backups import new_backup
+from frappe.utils.background_jobs import enqueue
+from frappe.integrations.offsite_backup_utils import get_latest_backup_file, send_email, validate_file_size, generate_files_backup
+from frappe.utils import cint, get_backups_path
+
+import requests
+import os
+from rq.timeouts import JobTimeoutException
+from urllib.parse import urlparse
+
+class NextCloudSettings(Document):
+
+	def validate(self):
+		if not self.enabled:
+			return
+
+@frappe.whitelist()
+def take_backup():
+	"""Enqueue longjob for taking backup to nextcloud"""
+	enqueue("frappe.integrations.doctype.nextcloud_settings.nextcloud_settings.start_backup", queue='long', timeout=1500)
+	frappe.msgprint(_("Queued for backup. It may take a few minutes to an hour."))
+
+def take_backups_daily():
+	take_backups_if("Daily")
+
+def take_backups_weekly():
+	take_backups_if("Weekly")
+
+def take_backups_if(freq):
+	if frappe.db.get_value("NextCloud Settings", None, "backup_frequency") == freq:
+		start_backup()
+
+def start_backup():
+	backup = NextCloudController()
+	backup.take_backup_nextcloud()
+
+class NextCloudController():
```

Comment:
Why the need for a separate class? You can have it in NextCloud Setting Class itself